Unity SDK docs, phase 1

timothymcmackin opened the pull request:
Preview: https://docs-staging-git-unity-phase-1-trili-tech.vercel.app/unity

Includes quickstart, overview, and reference info for main objects and prefabs. Also includes intro and setup information for the tutorial scenes.

<img width="285" alt="Screenshot 2024-01-11 at 1 11 39 PM" src="https://github.com/trilitech/tezos-developer-docs/assets/6494785/59b5c868-3fbe-4293-9cfb-7a5f8c6e35f1">

cnbrk13 commented on docs/dApps/unity/quickstart.md:
```diff
@@ -0,0 +1,373 @@
+
+1. Switch the wallet to Ghostnet and use the [Ghostnet faucet](https://faucet.ghostnet.teztnets.xyz/) to get some tez that you can use to send transactions to your Unity projects.
+
+1. Install the SDK from GitHub:
```
We should simply tell users to follow the installation steps on GitHub. That way we only have 1 place to maintain as things change.

timothymcmackin replied:
Left the installation section in but now it's a link to github and a mention of setting up a wallet.

cnbrk13 commented on docs/dApps/unity/quickstart.md:
```diff
@@ -0,0 +1,373 @@
+   If you copy these prefabs into your scene and run it, it shows a QR code that Tezos wallet applications can scan to connect with the application.
+   You can access these features through the prefab and change how the project manages its connection to users' wallets.
+
+   For example, the `QRCode` object receives a handshake from the `TezosManager` object and uses the handshake information to generate a URI and encode that URI to a QR code image:
```
I think we should skip QR code information as it's a bit confusing for newcomers.

cnbrk13 commented on docs/dApps/unity/quickstart.md:
```diff
@@ -0,0 +1,373 @@
+
+1. Add features to your project to use the connected account.
+For example, the `TezosSDK/Examples/Common/Scripts/AccountInfoUI.cs` file responds to the `AccountConnected` event, which runs when the user scans the QR code and approves the connection in their wallet app.
+You can use this event to get the address of the connected account, as in these steps:
```
We can skip these steps and just show the code below. We assume good Unity knowledge for readers, and we want to keep the Quickstart as concise as possible.

timothymcmackin replied:
Trimmed this section.

cnbrk13 commented on docs/dApps/unity/quickstart.md:
```diff
@@ -0,0 +1,373 @@
+          TezosManager.Instance.MessageReceiver.AccountDisconnected += OnAccountDisconnected;
+      }
+
+      private void OnAccountDisconnected(AccountInfo accountInfo)
```
This should be OnAccountConnected

cnbrk13 commented on docs/dApps/unity/quickstart.md:
````diff
@@ -0,0 +1,373 @@
+   }
+   ```
+
+   For a complete example, see the Transfer example scene.
````
This should be a link to the page.

timothymcmackin replied:
I'll link it when we have a separate page for the scenes.

cnbrk13 reviewed (commented):
I've made some minor adjustments all around. 

I'm also changing all the mentions of "example scenes" to "tutorial scenes" and "account" to "wallet" to standardise these keywords.

timothymcmackin commented on docs/unity.md:
```diff
@@ -0,0 +1,103 @@
+You can use Tezos via the SDK to:
+
+- Use a player's Tezos account as their account for a game and their wallet as their way of logging in to the game
+- Accept payments from players in tez
```
We don't capitalize "tez" in general.

timothymcmackin commented on docs/unity.md:
```diff
@@ -0,0 +1,103 @@
+
+   1. Open a scene in your project or create a scene.
+   1. In the Project panel, expand **Assets > TezosSDK > Runtime > Prefabs**.
+   1. From the Prefabs folder in the Project panel, drag the MainThreadExecutor prefab to the Hierarchy panel.
```
Removed reference to this prefab.

timothymcmackin commented on docs/unity.md:
```diff
@@ -0,0 +1,103 @@
+   1. Open a scene in your project or create a scene.
+   1. In the Project panel, expand **Assets > TezosSDK > Runtime > Prefabs**.
+   1. From the Prefabs folder in the Project panel, drag the MainThreadExecutor prefab to the Hierarchy panel.
+   1. In the same way, drag the TezosManager prefab to the Hierarchy panel.
```
Resolved with previous comment.

timothymcmackin commented on docs/unity.md:
```diff
@@ -0,0 +1,103 @@
+
+1. To enable copy and paste in the WebGL build, double-click the `WebGLCopyAndPaste.unitypackage` package, which is in the `TezosSDK/WebGLFrontend/output` folder of the SDK, to install it.
+
+   This package includes the script `WebGLCopyAndPaste.cs` alongside with `StreamingAssets` and `WebGLTemplates` folders inside your project Assets directory.
```
"This package automatically enables copy and paste..."

timothymcmackin commented on docs/unity.md:
```diff
@@ -0,0 +1,103 @@
+      This prefab provides prerequisites to use Tezos in a scene.
+      Its fields control what users see in their wallet applications before connecting to the project.
+
+   1. If you want to publish the project to WebGL, follow the steps in [Enabling WebGL support](#enabling-webgl-support).
```
No need to bold that text.

timothymcmackin commented on docs/unity.md:
```diff
@@ -0,0 +1,103 @@
+
+<!-- TODO: Link to complete reference for these objects -->
+
+## Dependencies
```
> The SDK also uses the [Newtonsoft JSON Unity Package] ...

timothymcmackin commented on docs/unity/scenes.md:
```diff
@@ -0,0 +1,226 @@
+---
+title: Unity SDK example scenes
```
Making this "Unity SDK tutorial scenes" in lowercase for consistency with other page titles.

timothymcmackin commented on docs/unity/quickstart.md:
````diff
@@ -0,0 +1,319 @@
+To get the tokens that the connected account owns, call the [`API.GetTokensForOwner()`](./reference/API#gettokensforowner) method in a coroutine.
+This example prints information about the tokens that the account owns to the log:
+
+```csharp
+using Beacon.Sdk.Beacon;
+using TezosSDK.Beacon;
+using TezosSDK.Tezos;
+using TezosSDK.Helpers;
+using TezosSDK.Tezos.API.Models.Tokens;
+using TezosSDK.Tezos.API.Models.Filters;
+using System.Linq;
+
+// ...
+
````
ok, removing these imports because a C# or Unity dev should know where to find the objects. I put them in there mostly because I didn't know an easy way to look up the imports when I copied a code block.

cnbrk13 replied:
Okay nice 👍  Most IDEs 99% of the time find and add these for you anyway. And if not, as you said. This code block is already too long, nice to shorten it.

timothymcmackin commented:
> I've made some minor adjustments all around.
> 
> I'm also changing all the mentions of "example scenes" to "tutorial scenes" and "account" to "wallet" to standardise these keywords.

Thanks for the review. I addressed your comments and got these globals out of them; I'll go over the docs with these in mind:

- Make sure code in other places is consistent with changes to quickstart code
- Removal of MainThreadExecutor prefab
- "example scenes" are now "tutorial scenes"
- Remove imports from code examples
- WalletConnection scene: Almost all of this is done via TezosAuthenticator now, so we need to talk less about the scene and more about TezosAuthenticator.

timothymcmackin commented on docs/unity/scenes.md:
```diff
@@ -0,0 +1,226 @@
+Before using any of the scenes, install a Tezos-compatible wallet on a mobile device that has a camera and can scan QR codes and get some test tez tokens that you can use to pay transaction fees.
+For instructions, see [Installing and funding a wallet](./developing/wallet-setup).
+
+## WalletConnection scene
```
Added some more info about TezosAuthenticator.

cnbrk13 commented on docs/unity/reference/EventManager.md:
```diff
@@ -0,0 +1,86 @@
+Runs when a wallet connects successfully.
+Returns a `TezosSDK.Beacon.WalletInfo` object with information that includes the address of the connected account.
+
+### `public event Action<ErrorInfo> AccountConnectionFailed`
+
+Runs when a connection to an account fails.
```
```suggestion
### `public event Action<ErrorInfo> WalletConnectionFailed`

Runs when a connection to a wallet fails.
```
